scripts/source_code.py

```diff
@@ -152,6 +152,7 @@ def py_get_func_code_demangled(
     "c": clang_get_func_code,
     "cpp": clang_get_func_code_mangled,
     "cc": clang_get_func_code_mangled,
+    "cxx": clang_get_func_code_mangled,
     "python": py_get_func_code_demangled,
```

Comment:
Please double check if this dict is a `proj_language` or `file_extension` to extractor mapping. If the key is file extension, we can change "python" to "py".

Reply:
I think the usage of the dictionary right now is inconsistent. Hongxiang changed the demangle.py script to use the project language, which I think would work fine with python projects since all of the source code for those projects should be in python. C++ is the only exception so far since some source code could be in C and some in C++.

Reply:
can you also put a summarization of this in a comment above the dict? thanks